connectors/mongo/parallel_writer.go

```diff
@@ -140,6 +143,15 @@ func (bwa *ParallelWriter) ScheduleBarrier(barrierMsg iface.DataMessage) error {
 	}
 
 	bwa.BroadcastMessage(barrierMsg)
+
+	if barrierMsg.BarrierType == iface.BarrierType_Block {
+		slog.Debug("Blocking barrier encountered.")
+		for i := 0; i < bwa.numWorkers; i++ {
+			<-bwa.blockBarrier
```

Comment:
Wouldn't that require N messages in the blockBarrier channel? (for each worker)

Reply:
n messages total (1 per worker). this is blocking on the parallel writer itself, not the writer worker

Reply:
Got it